packages/lib/src/vis/shared/VisCanvas.tsx

```diff
@@ -79,21 +80,16 @@ function VisCanvas(props: PropsWithChildren<Props>) {
               <ThresholdAdjuster value={raycasterThreshold} />
             )}
           </VisCanvasProvider>
-
-          <Html>
-            <svg
-              ref={(elem) => setSvgOverlay(elem || undefined)}
-              className={styles.svgOverlay}
-            />
-          </Html>
-          <Html>
-            <div
-              ref={(elem) => setFloatingToolbar(elem || undefined)}
-              className={styles.floatingToolbar}
-            />
-          </Html>
         </R3FCanvas>
+
+        <svg
+          ref={setSvgOverlay}
+          className={styles.svgOverlay}
+          overflow="hidden"
```

Comment:
User agents set `overflow: hidden` on `svg` elements that are not document roots ([source](https://developer.mozilla.org/en-US/docs/Web/SVG/Attribute/overflow); this is why we have to set `overflow: visible` on the axes and the grid), but I prefer it to be explicit here.